languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -150,6 +150,13 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
     edge @class.type -> ref
 }
 
+(class_declaration
+  superclass: (superclass
+    (generic_type
+      (type_identifier)) @superclass_name)) @class {
+    edge @superclass_name.lexical_scope -> @class.lexical_scope
+}
```

Comment:
I honestly could not get the syntax for `?` to work for having a nested optional `generic_type` in the rule above. If you want to take a stab at it and push up an update or pair at all tomorrow, I'm happy to give it another shot. I agree it would be cleaner that way!